File: sdk/javascript/examples/per-call.mjs
// Per-call client: every read is one request to the server. Good for scripts,
// jobs, and serverless. Run with:  FLAG_IT_SDK_KEY=sdk-… node per-call.mjs
import { createClient, context } from '@flag-it/sdk'

const client = createClient({
  baseUrl: process.env.FLAG_IT_URL ?? 'http://localhost:8080',
  sdkKey: process.env.FLAG_IT_SDK_KEY,
})

const user = context('user', 'u-123', { plan: 'pro' })

// Typed read — never throws, returns the fallback on error / type mismatch:
console.log('new-dashboard:', await client.boolVariation('new-dashboard', user, false))

// Full evaluation — value + variation index + reason (throws on a network error):
const ev = await client.evaluate('new-dashboard', user)
console.log('detail:', ev.value, ev.variation, ev.reason)

// Everything at once — one round-trip, good for bootstrapping a page:
const all = await client.allFlags(user)
console.log('all flags:', Object.keys(all))

File: sdk/python/examples/per_call.py
"""Per-call client: every read is one request to the server. Good for scripts,
jobs, and serverless. Run with:  FLAG_IT_SDK_KEY=sdk-… python per_call.py
"""

import os

import flagit

client = flagit.Client(
    base_url=os.environ.get("FLAG_IT_URL", "http://localhost:8080"),
    sdk_key=os.environ["FLAG_IT_SDK_KEY"],
)
user = flagit.context("user", "u-123", {"plan": "pro"})

# Canonical read (like LaunchDarkly) — never raises, returns the default on error:
print("new-dashboard:", client.variation("new-dashboard", user, False))

# With explanation — value + variation index + reason:
detail = client.variation_detail("new-dashboard", user, False)
print("detail:", detail.value, detail.variation_index, detail.reason)

# Everything at once — one round-trip, good for bootstrapping:
state = client.all_flags_state(user)
print("all flags:", list(state.keys()))
print("values:", state.to_values_dict())
